services/src/main/scala/cromwell/services/metadata/impl/MetadataDatabaseAccess.scala

```diff
@@ -125,6 +125,13 @@ trait MetadataDatabaseAccess {
       case MetadataQuery(_, None, None, Some(includeKeys), Some(excludeKeys), _) => Future.failed(
         new IllegalArgumentException(
           s"Include/Exclude keys may not be mixed: include = $includeKeys, exclude = $excludeKeys"))
+      case MetadataQuery(_, None, Some(_), None, Some(_), _)    |
+           MetadataQuery(_, None, Some(_), Some(_), None, _)    |
+           MetadataQuery(_, None, Some(_), Some(_), Some(_), _) |
+           MetadataQuery(_, Some(_), None, Some(_), Some(_), _) |
+           MetadataQuery(_, Some(_), Some(_), None, Some(_), _) |
+           MetadataQuery(_, Some(_), Some(_), Some(_), None, _) |
+           MetadataQuery(_, Some(_), Some(_), Some(_), Some(_), _) => Future.failed(new IllegalStateException("Invalid MetadataQuery"))
```

Comment:
1) sure
2) Presumably it'd satisfy the compiler, so sure. I just copy/pasted that from the error message w/o thinking about it too much (other than 'yuck') :)